projects/novo-elements/src/elements/data-table/data-table-expand.directive.ts

```diff
@@ -17,11 +17,13 @@ export class NovoDataTableExpandDirective<T> implements OnDestroy {
   private subscription: Subscription;
 
   constructor(public vcRef: ViewContainerRef, private state: DataTableState<T>, private dataTable: NovoDataTable<T>) {
-    this.subscription = this.state.expandSource.subscribe(() => {
-      if (dataTable.isExpanded(this.row)) {
-        this.render();
-      } else {
-        this.clear();
+    this.subscription = this.state.expandSource.subscribe((targetId?: number) => {
+      if ([undefined, ((this.row as unknown) as { id: number }).id].includes(targetId)) {
+        if (dataTable.isExpanded(this.row)) {
```

Comment:
why is undefined explicitly called out here?

Reply:
@chris-erxleben @amrutha-rajiv I pushed some changes to make this more straightforward. LMK what you think and resolve if you think it's suitable.